docs-new/.github/workflows/ci.yml

```diff
@@ -0,0 +1,29 @@
+name: ci 
+on:
+  push:
+    branches:
+      - master 
```

Comment:
I think this is an issue:

> Adjust the structure to match the layout of the website itself, where ./home/*.md contains the evergreen content and ./home/docs/<version>/*.md contains the versions of Iceberg.

This would mean that it shows features that haven't been released yet. I think it is great to build these pages, but we have to clearly state that it is the dev version. I think Arrow is a nice example:

![image](https://github.com/apache/iceberg/assets/1134248/2b2bb84c-2239-4715-a1f5-65ce7ad089f0)

Reply:
> This would mean that it shows features that haven't been released yet. 

Not on the live site. I assumed that this build would only trigger on a release, so (latest) and the current version (e.g. 1.3.1 today) would just be duplicate docs (Unfortunately redirects from `latest` > `<version>` that maintain the `/docs/latest/` url would require some fancy DNS records that @rdblue said we should avoid for now).

Later, we should defnitely consider adding a (nightly) or (dev) build and we would label that as such. For now, just assume that we only build this during a release. Unless the concern is around when people build the site locally.

Reply:
Alright, then you should not run the workflow on a `push` event on the main/master branch.

Then you want to replace it with:

```yaml
on:
    tags:        
      - apache-iceberg-*
```


Also, in PyIceberg we have it as a [step in the release](https://py.iceberg.apache.org/how-to-release/#release-the-docs):

https://github.com/apache/iceberg/blob/eaf7c4f2e2adc952b5a150cf0b854adebfb9fdf0/.github/workflows/python-ci-docs.yml#L21-L22

Reply:
That makes sense. Yeah, let's hit this deployment on the next PR.